### DISCUSSION TITLE
OpsTrace as an Keptn SLI provider

### DISCUSSION THREAD

**Discussion by agardnerIT** [opstrace/opstrace#1208] [Ideas]
Hi team, just seen your [DoK talk](https://www.youtube.com/watch?v=eVZ-l1gRbrc) and as a Keptn contributor, it got me thinking - can we use OpsTrace as an SLI provider for Keptn?

From [the docs](https://keptn.sh/docs/0.8.x/integrations/sli_provider/):
```
An SLI-provider is an implementation of a Keptn-service with a dedicated purpose.
This type of service is responsible for querying an external data source for SLIs that are
then used by Keptn to evaluate an SLO.
```

If we can built this, you get Opstrace powered quality gates, SLO-based pipelines and auto-remediation in production powered by OpsTrace + Keptn.
 
Happy to work with you on building this integration.

**Comment by triclambert** [comment]
Hi Adam, thanks for reaching out!

We love open APIs and open standards, and we love to integrate with the broader ecosystem.  Opstrace exposes a (secure-by-default) Prometheus API (for remote_write and remote_read).  Your existing [external Prometheus](https://github.com/keptn-contrib/prometheus-service#using-an-external-prometheus-instance) may already solve half the problem, however, we require a [bearer token as authentication proof](https://opstrace.com/docs/guides/user/sending-metrics-with-prometheus#choosing-bearer_token_file) instead of basic auth.  

We'll think on this a bit and get back to you.
